libfeed/feed-proxy-factory.c

```diff
@@ -0,0 +1,390 @@
+                                        GCancellable *cancellable)
+{
+  InstantiateProxyForInterfaceData *data = task_data;
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(EosDiscoveryFeedKnowledgeAppProxy) ka_proxy =
+    instantiate_proxy_for_interface_sync (data->connection,
+                                          data->provider_info,
+                                          data->interface_name,
+                                          data->interface_metadata,
+                                          cancellable,
+                                          &local_error);
+
+  if (ka_proxy == NULL)
+    {
+      g_task_return_error (task, g_steal_pointer (&local_error));
+      return;
+    }
+
+  g_task_return_pointer (task, g_steal_pointer (&ka_proxy), g_object_unref);
+}
+
+static void
+instantiate_proxy_for_interface (GDBusConnection              *connection,
+                                 EosDiscoveryFeedProviderInfo *provider_info,
+                                 const gchar                  *interface_name,
+                                 const gchar                  *interface_metadata,
+                                 GCancellable                 *cancellable,
+                                 GAsyncReadyCallback           callback,
+                                 gpointer                      user_data)
+{
+  g_autoptr(GTask) task = g_task_new (NULL, cancellable, callback, user_data);
+
+  g_task_set_return_on_cancel (task, TRUE);
+  g_task_set_task_data (task,
+                        instantiate_proxy_for_interface_data_new (connection,
+                                                                  provider_info,
+                                                                  interface_name,
+                                                                  interface_metadata),
+                        (GDestroyNotify) instantiate_proxy_for_interface_data_free);
+  g_task_run_in_thread (task, instantiate_proxy_for_interface_thread);
+}
+
+static void
+on_received_all_instantiation_results (GObject      *source,
+                                       GAsyncResult *result,
+                                       gpointer      user_data)
+{
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(GTask) task = G_TASK (user_data);
+  g_autoptr(GPtrArray) instantiation_results = g_task_propagate_pointer (G_TASK (result),
+                                                                         &local_error);
+  g_autoptr(GPtrArray) proxies = NULL;
+  guint i = 0;
+
+  if (instantiation_results == NULL)
+    {
+      g_task_return_error (task, g_steal_pointer (&local_error));
+      return;
+    }
+
+  proxies = g_ptr_array_new_full (instantiation_results->len, g_object_unref);
+
+  /* Go through each of the results, complain about the ones that failed
+   * but keep the ones that sucessfuly returned a proxy. */
+  for (i = 0; i < instantiation_results->len; ++i)
```

Comment:
It is, my convention here is to skip the loop initializer if it comes right after the declaration + initialization, otherwise to reinitialize it again for clarity.